### PR TITLE
EPMRPP-37910 || Launch retry flag api

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/item/impl/StartTestItemHandlerImpl.java [hunk 1]
```diff
@@ -152,6 +154,9 @@ private void validate(ReportPortalUser user, ReportPortalUser.ProjectDetails pro
 	 * @param parent Parent item
 	 */
 	private void validate(StartTestItemRQ rq, TestItem parent) {
+
+		//		expect(parent.getRetryOf(), isNull()::test).verify(UNABLE_TO_SAVE_CHILD_ITEM_FOR_THE_RETRY, parent.getItemId());
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/StartTestItemHandlerImpl.java]: ?

File: src/main/java/com/epam/ta/reportportal/core/item/impl/DeleteTestItemHandlerImpl.java [hunk 2]
```diff
@@ -52,26 +53,41 @@
...
+		Optional<TestItem> retryParent = ofNullable(item.getRetryOf()).map(retryParentId -> testItemRepository.findById(retryParentId)
+				.orElseThrow(() -> new ReportPortalException(ErrorType.TEST_ITEM_NOT_FOUND, retryParentId)));
+
+		Launch launch = retryParent.map(TestItem::getLaunch).orElseGet(item::getLaunch);
+
+		validate(item, launch, reportPortalUser, projectDetails);
 		Optional<TestItem> parent = ofNullable(item.getParent());
-		testItemRepository.delete(item);
 
+		retryParent.ifPresent(rp -> rp.getRetries().remove(item));
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/DeleteTestItemHandlerImpl.java]: we can't delete retry items
they should be deleted only when parent is

File: src/main/java/com/epam/ta/reportportal/core/item/impl/StartTestItemHandlerImpl.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/item/impl/StartTestItemHandlerImpl.java]: fixed

File: src/main/java/com/epam/ta/reportportal/core/item/impl/DeleteTestItemHandlerImpl.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/item/impl/DeleteTestItemHandlerImpl.java]: fixed

**Pull request merged**